ITISFoundation/osparc-simcore
# ✨ Maintenance: disable Traefik sticky connection and Socket.IO long polling 🚨

## What do these changes do?
We are in 2022, websockets are available everywhere:
- no need for socket.io long polling feature
- therefore no need to continue with having sticky connection defined on traefik (this ensure one client always gets responses from the same webserver)

Removing connection stickyness means that all webserver instances can reply to a client, instead of only one, thus this should improve responsiveness.

Reference on socket.io: https://socket.io/docs/v4/using-multiple-nodes#enabling-sticky-session


This works because currently the webserver instances are all receiving the logs/progress through rabbitMQ (very innefficient, and will be one of the next changes, so keep the hope)



## Related issue/s




## How to test


1. connect to osparc with debugging tools open
2. check the websocket properly connected
3. use osparc as usual (run computational services, dynamic ones)
4. check P2P responsivity tests, there might be a slight increase in RPS

## Checklist
